speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -535,13 +553,36 @@ static void SelectBlackBox(void* /* dummy */)
     // Visual feedback of choice
     std::string buttonText = MSG_BLACK_BOX_NORMAL_TEXT + path.filename().string();
     GfuiButtonSetText(s_scrHandle, m_blackBoxButtonControl, buttonText.c_str());
-    GfuiLabelSetText(s_scrHandle, m_errorLabel, MSG_ONLY_HINT);  // Reset error label
+    // Reset error label; but only if we can guarantee the label isn't being used by the user id
+    if (m_userIdChosen)
+    {
+        GfuiLabelSetText(s_scrHandle, m_errorLabel, MSG_ONLY_HINT);
+    }
 
     // Only after validation copy into the actual variable
     strcpy_s(m_blackBoxFilePath, BLACKBOX_PATH_SIZE, buf);
     m_blackBoxChosen = true;
 }
 
+/// @brief Generates a random user id
+static void GenerateUid(void* /* dummy */)
+{
+    // Create random userId
```

Comment:
Add your comments above a block of code, but not above every single line. In my opinion no comments are needed at all for those lines, everyone will be able to quickly understand what the code does by the code itself.